contrib/automation_tests/test_cases/connection_window.py

```diff
@@ -95,13 +96,93 @@ def _execute(self):
         self.expect_true(instance_list_overlay.is_visible(), "Instance overlay is visible")
         logging.info('Found InstanceListOverlay and its visible')
 
-        window = self.suite.top_window()
-        window.InstanceList.DataItem0.wait('exists', timeout=100)
-        self.expect_true(instance_list_overlay.rectangle().width() == 0,
-                         "InstanceOverlay has width() 0 (is not shown)")
+        wait_for_condition(
+            lambda: self.find_control('Group', 'InstanceListOverlay', raise_on_failure=False) is
+            None, 100)
         logging.info('Loading done, overlay is hidden')
-        logging.info('Found %s rows in the instance list', instance_list.item_count())
-        self.expect_true(instance_list.item_count() >= 1, 'Found at least one instance')
+        self.expect_true(get_number_of_instances_in_list(self) >= 1, 'Found at least one instance')
+
+
+class SelectProject(E2ETestCase):
+    """
+    Expand the project selection combo box and click the entry `project_name`. Also waits 
+    appropriately when loading takes place.
+    """
+
+    def _execute(self, project_name: str):
+        logging.info('Select project with name ' + project_name)
+        project_combo_box = self.find_control(name='ProjectSelectionComboBox')
+        # wait until loading is done
+        wait_for_condition(lambda: project_combo_box.is_enabled() is True, 25)
+        project_combo_box.click_input()
+
+        default_project = self.find_combo_box_item(text=project_name)
+        logging.info('Expanded project combo box and found entry ' + project_name)
+        default_project.click_input()
+        # After project changing, loading takes place. Wait until the overlay is hidden
+        wait_for_condition(
+            lambda: self.find_control('Group', 'InstanceListOverlay', raise_on_failure=False) is
+            None, 100)
+        logging.info('Successfully selected project ' + project_name +
+                     'and waited until loading is done')
+
+
+class SelectNextProject(E2ETestCase):
+    """
+    Expand the project selection combo box and select the next project by pressing the down arrow
+    and enter. Also waits appropriately when loading takes place.
+    """
+
+    def _execute(self):
```

Comment:
Same comment as above: Could this test verify that changing the project actually have the expected effect? Simple smoke test could be that the list of instances actually changed.

Reply:
Its not super trivial to check whether the list of instances changed. It would require some code to get the list of instances with its properties and code to compare lists. I will not do this in this change, maybe I find time in the fixit week